tribe-ext-formatted-event-date-shortcode/index.php

```diff
@@ -106,15 +109,119 @@ public function get_event_timestamp( $event_date_as_string ) {
+			return false;
+		}
+		
+		$datetime_format_constants = array(
+			'DATE_ATOM',
+			'DATE_COOKIE',
+			'DATE_ISO8601', // Note: This format is not compatible with ISO-8601, but is left this way for backward compatibility reasons. Use DATE_ATOM for compatibility with ISO-8601 instead.
+			'DATE_RFC822',
+			'DATE_RFC850',
+			'DATE_RFC1036',
+			'DATE_RFC1123',
+			'DATE_RFC2822',
+			'DATE_RFC3339', // Same as DATE_ATOM (since PHP 5.1.3) 
+			'DATE_RSS',
+			'DATE_W3C',
+		);
 		
+		if ( in_array( $format, $datetime_format_constants ) ) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	
+	/**
+	 * Filter timezone_string option and run date_i18n(), then remove the filter on timezone_string option
+	 *
+	 * We do things this way in order to allow the shortcode's date format argument to properly display
+	 * when the event's timezone or the shortcode's timezone is not the same as the WordPress
+	 * timezone option... while still enabling translation of things like month name according
+	 * to WordPress' locale/language setting. Note, however, that date_i18n()'s locale/translation
+	 * juggling does not translate timezone text (e.g. "EST" or "America/New_York").
+	 *
+	 * @since 1.1
+	 * @access private
+	 *
+	 * @link https://developer.wordpress.org/reference/functions/get_option/
+	 * @link https://developer.wordpress.org/reference/functions/date_i18n/
+	 *
+	 * @return bool|string
+	 */
+	private function date_i18n_custom_timezone( $format, $timestamp ) {
+		if ( empty( $format ) || empty( $timestamp ) ) {
+			return false;
+		}
+		
+		$timezone = $this->use_this_timezone();
+		
+		// if no specific timezone, use WP's default from date_i18n(), else use the passed timezone but still run through date_i18n() to get the translation benefits
+		if ( empty( $timezone ) ) {
+			$result = date_i18n( $format, $timestamp );
+		} else {
+			// set timezone so that date_i18n() works correctly
+			$orig_timezone = date_default_timezone_get(); // will return 'UTC' as a fallback but could potentially be a TZ environment variable
+
+			if ( $orig_timezone != $timezone ) {
+				date_default_timezone_set( $timezone );
+				$revert_timezone = true;
+			}
+			
+			// get the date
+			add_filter( 'option_timezone_string', array( $this, 'use_this_timezone' ) );
+			$result = date_i18n( $format, $timestamp );
+			remove_filter( 'option_timezone_string', array( $this, 'use_this_timezone' ) );
+			
+			// set the timezone back to what it was
+			if ( ! empty( $revert_timezone ) ) {
+				date_default_timezone_set( $orig_timezone );
+			}
+		}
+				
+		return $result;
+	}
+
+	/**
+	 * Return this class' timezone, set via the shortcode's output method
+	 *
+	 * @since 1.1
+	 *
+	 * @see $this->shortcode_output()
+	 *
+	 * @return string
+	 */
+	public function use_this_timezone() {
+		$timezone = $this->timezone_for_display;
+		
+		// only allow valid timezones
+		if ( ! in_array( $timezone, timezone_identifiers_list() ) ) {
+			$timezone = '';
```

Comment:
I think this should also be set to false. It looks like the option this filters expects `false`, or a valid timezone.